strawberryfields/decompositions.py

```diff
@@ -416,7 +416,7 @@ def mach_zehnder(m, n, internal_phase, external_phase, nmax):
     BS[m, n] = 1.0j / np.sqrt(2)
     BS[n, m] = 1.0j / np.sqrt(2)
     BS[n, n] = 1.0 / np.sqrt(2)
-    return BS @ Rinternal @ BS @ Rexternal
+    return np.real_if_close(BS @ Rinternal @ BS @ Rexternal)
```

Comment:
Fixes a bug (?) introduced in NumPy 1.19, which results in significant complex rounding error. If not removed, this complex rounding error propagates throughout all the decompositions, producing incorrect results.